tests/unit/plugins/module_utils/test_cmd_runner.py

```diff
@@ -0,0 +1,303 @@
+
+
+TC_FORMATS = dict(
+    simple_boolean__true=(fmt.as_bool, ("--superflag",), True, ["--superflag"]),
+    simple_boolean__false=(fmt.as_bool, ("--superflag",), False, []),
+    simple_boolean__none=(fmt.as_bool, ("--superflag",), None, []),
+    simple_boolean_not__true=(fmt.as_bool_not, ("--superflag",), True, []),
+    simple_boolean_not__false=(fmt.as_bool_not, ("--superflag",), False, ["--superflag"]),
+    simple_boolean_not__none=(fmt.as_bool_not, ("--superflag",), None, ["--superflag"]),
+    simple_optval__str=(fmt.as_optval, ("-t",), "potatoes", ["-tpotatoes"]),
+    simple_optval__int=(fmt.as_optval, ("-t",), 42, ["-t42"]),
+    simple_opt_val__str=(fmt.as_opt_val, ("-t",), "potatoes", ["-t", "potatoes"]),
+    simple_opt_val__int=(fmt.as_opt_val, ("-t",), 42, ["-t", "42"]),
+    simple_opt_eq_val__str=(fmt.as_opt_eq_val, ("--food",), "potatoes", ["--food=potatoes"]),
+    simple_opt_eq_val__int=(fmt.as_opt_eq_val, ("--answer",), 42, ["--answer=42"]),
+    simple_list_potato=(fmt.as_list, (), "literal_potato", ["literal_potato"]),
+    simple_list_42=(fmt.as_list, (), 42, ["42"]),
+    simple_map=(fmt.as_map, ({'a': 1, 'b': 2, 'c': 3},), 'b', ["2"]),
+    simple_default_type__list=(fmt.as_default_type, ("list",), [1, 2, 3, 5, 8], ["--1", "--2", "--3", "--5", "--8"]),
+    simple_default_type__bool_true=(fmt.as_default_type, ("bool", "what"), True, ["--what"]),
+    simple_default_type__bool_false=(fmt.as_default_type, ("bool", "what"), False, []),
+    simple_default_type__potato=(fmt.as_default_type, ("any-other-type", "potato"), "42", ["--potato", "42"]),
+    simple_fixed_true=(fmt.as_fixed, [("--always-here", "--forever")], True, ["--always-here", "--forever"]),
+    simple_fixed_false=(fmt.as_fixed, [("--always-here", "--forever")], False, ["--always-here", "--forever"]),
+    simple_fixed_none=(fmt.as_fixed, [("--always-here", "--forever")], None, ["--always-here", "--forever"]),
+    simple_fixed_str=(fmt.as_fixed, [("--always-here", "--forever")], "something", ["--always-here", "--forever"]),
+)
+if tuple(version_info) >= (3, 1):
+    from collections import OrderedDict
+
+    # needs OrderedDict to provide a consistent key order
+    TC_FORMATS["simple_default_type__dict"] = (  # type: ignore
+        fmt.as_default_type,
+        ("dict",),
+        OrderedDict((('a', 1), ('b', 2))),
+        ["--a=1", "--b=2"]
+    )
+TC_FORMATS_IDS = sorted(TC_FORMATS.keys())
+
+
+@pytest.mark.parametrize('func, fmt_opt, value, expected',
+                         (TC_FORMATS[tc] for tc in TC_FORMATS_IDS),
+                         ids=TC_FORMATS_IDS)
+def test_arg_format(func, fmt_opt, value, expected):
+    fmt_func = func(*fmt_opt)
+    actual = fmt_func(value, ctx_ignore_none=True)
+    print("formatted string = {0}".format(actual))
+    assert actual == expected, "actual = {0}".format(actual)
+
+
+TC_RUNNER = dict(
+    # aa_bb=(
+    #     # input
```

Comment:
Why are these commented out?

Reply:
To serve as an example/sample with all the possible elements of the test case.

Reply:
Why not remove the comments so that it is actually running, to make sure it really works and isn't accidentally broken?

Reply:
That blob is not actually runnable, it has not been tested as such. The "runnable" versions are below ir.